src/replay_stage.rs

```diff
@@ -191,39 +191,34 @@ impl ReplayStage {
         let to_leader_sender = to_leader_sender.clone();
         let subscriptions_ = subscriptions.clone();
 
-        let (bank, last_entry_id) = {
+        let (bank, last_entry_id, mut current_blob_index) = {
             let mut bank_forks = bank_forks.write().unwrap();
             bank_forks.set_working_bank_id(bank_forks_info[0].bank_id);
-            (bank_forks.working_bank(), bank_forks_info[0].last_entry_id)
+            (
+                bank_forks.working_bank(),
+                bank_forks_info[0].last_entry_id,
+                bank_forks_info[0].entry_height,
+            )
         };
         let last_entry_id = Arc::new(RwLock::new(last_entry_id));
-
-        let mut current_blob_index = {
-            let leader_scheduler = leader_scheduler.read().unwrap();
-            let slot = leader_scheduler.tick_height_to_slot(bank.tick_height() + 1);
-
-            let leader_id = leader_scheduler
-                .get_leader_for_slot(slot)
-                .expect("Leader not known after processing bank");
-            trace!("node {:?} scheduled as leader for slot {}", leader_id, slot,);
-
-            // Send a rotation notification back to Fullnode to initialize the TPU to the right
-            // state
-            to_leader_sender
-                .send(TvuRotationInfo {
-                    bank: Bank::new_from_parent(&bank, &leader_id),
-                    last_entry_id: *last_entry_id.read().unwrap(),
-                    slot,
-                    leader_id,
-                })
-                .unwrap();
-
-            blocktree
-                .meta(slot)
-                .expect("Database error")
-                .map(|meta| meta.consumed)
-                .unwrap_or(0)
-        };
+        let leader_scheduler = leader_scheduler.read().unwrap();
```

Comment:
oh this also wrapped in a block to release the leader_scheduler read lock

Reply:
hehe, test just died yelling at me for that xD

Reply:
excellent!